src/pipeline_benchmark/benchmark.py

```diff
@@ -225,65 +324,29 @@ def judge_pipeline(expected_coupons: List[Coupon],
 
 
 def run_pipeline(pipeline_command: str,
-                 input_folder: str,
-                 is_new_format: bool = False,
-                 is_simple: bool = False) -> Optional[List[Coupon]]:
+                 input: str,
+                 is_simple: bool) -> List[Union[Coupon, CouponSimple]]:
     """
     This function will run the pipeline with the input data and return the
-    generated coupons. The function will return None if the pipeline fails to run. 
-    The result of the pipeline must be written to a file called output.json.
+    generated coupons.
     
     :param pipeline_command: The command to run the pipeline
-    :param input_folder: The path to the folder with the input data
-    :param is_new_format: A boolean value that indicates if the input data is in the
-                        new format (default is False)
-    :param is_simple: A boolean value that indicates if the coupons are in the
-                    simple format (default is False)
-    :return: A list of Coupon objects that represent the generated coupons
-            or None if the pipeline fails to run
-    """
-
-    if os.path.exists(OUTPUT_FILE):
-        os.remove(OUTPUT_FILE)
-
-    if is_new_format:
-        input_file = os.path.join(input_folder, INPUT_FILE)
-        pipeline_command = pipeline_command + " < " + input_file + " > " + OUTPUT_FILE
-    else:
-        pipeline_command = pipeline_command + " > " + OUTPUT_FILE
-    print(f"Running the pipeline with the command: {pipeline_command}")
-
-    try:
-        subprocess.run(pipeline_command, shell=True, check=True)
+    :param input: The path to the file with the input data
+    :param is_simple: A boolean value that indicates if the coupons are in the simple format
+    :return: A list of Coupon or CouponSimple objects generated by the pipeline
 
-        # The pipeline must write the output to a file called output.json
-        proto_coupons = []
-        with open(OUTPUT_FILE, "r") as file:
-            coupons = json.load(file)
+    :raises ValueError: If the output file is not in the expected format
+    :raises subprocess.CalledProcessError: If the pipeline fails to run
+    """
 
-        for coupon in coupons:
-            if is_simple:
-                proto_coupons.append(
-                    CouponSimple(product_name=coupon["name"],
-                                 discount_text=coupon["text"],
-                                 validity_text=coupon["validity"]))
-                continue
-
-            proto_coupons.append(
-                Coupon(product_name=coupon["product_name"],
-                       new_price=coupon["new_price"],
-                       old_price=coupon["old_price"],
-                       percents=coupon["percents"],
-                       other_discounts=coupon["other_discounts"],
-                       dates=coupon["dates"]))
-        return proto_coupons
+    pipeline_command = pipeline_command + " < " + input + " > " + OUTPUT_FILE
```

Comment:
i suggest adding quotation marks to handle filepaths with spaces
```suggestion
    pipeline_command = pipeline_command + " <\" " + input + "\" > " + OUTPUT_FILE
```

Reply:
good point

Reply:
done